src/test/java/software/amazon/awssdk/crt/test/SigningTest.java

```diff
@@ -272,6 +275,61 @@ public Credentials getCredentials() {
         assertTrue(failed);
     }
 
+    /**
+     * Tests that signing is successful and doesn't add any headers when credentials are anonymous.
+     * @throws Exception
+     */
+    @Test
+    public void testSigningAnonymousCredentials() throws Exception {
+            HttpRequest request = createSimpleRequest("https://www.example.com", "POST", "/derp", "<body>Hello</body>");
+
+            try (AwsSigningConfig config = new AwsSigningConfig()) {
+                config.setAlgorithm(AwsSigningConfig.AwsSigningAlgorithm.SIGV4);
+                config.setSignatureType(AwsSigningConfig.AwsSignatureType.HTTP_REQUEST_VIA_HEADERS);
+                config.setRegion("us-east-1");
+                config.setService("service");
+                config.setTime(System.currentTimeMillis());
+                config.setUseDoubleUriEncode(true);
+                config.setShouldNormalizeUriPath(true);
+                config.setSignedBodyValue(AwsSigningConfig.AwsSignedBodyValue.EMPTY_SHA256);
+                config.setCredentials(Credentials.createAnonymousCredentials());
+
+                CompletableFuture<HttpRequest> result = AwsSigner.signRequest(request, config);
+                HttpRequest signedRequest = result.get();
+                assertNotNull(signedRequest);
+                assertEquals(request.getHeaders().size(), signedRequest.getHeaders().size());
+            }
+    }
+
+    /**
+     * Tests that an exception is thrown when credentials are null.
+     * @throws Exception
+     */
+    @Test(expected = CrtRuntimeException.class)
+    public void testSigningFailNullCredentials() throws Exception {
+        HttpRequest request = createSimpleRequest("https://www.example.com", "POST", "/derp", "<body>Hello</body>");
+        try{
+            AwsSigningConfig config = new AwsSigningConfig();
+            config.setAlgorithm(AwsSigningConfig.AwsSigningAlgorithm.SIGV4);
+            config.setSignatureType(AwsSigningConfig.AwsSignatureType.HTTP_REQUEST_VIA_HEADERS);
+            config.setRegion("us-east-1");
+            config.setService("service");
+            config.setTime(System.currentTimeMillis());
+            config.setUseDoubleUriEncode(true);
+            config.setShouldNormalizeUriPath(true);
+            config.setSignedBodyValue(AwsSigningConfig.AwsSignedBodyValue.EMPTY_SHA256);
+
+            CompletableFuture<HttpRequest> result = AwsSigner.signRequest(request, config);
+            HttpRequest signedRequest = result.get(); //This should throw an exception
+        } catch (ExecutionException e){
```

Comment:
minor: im not a big fan of rethrowing exception in the test to make sure its matches the type. 
what about using whenComplete and checking if it returns exception or result?